openhands/core/config/app_config.py

```diff
@@ -44,6 +44,9 @@ class AppConfig:
         file_uploads_allowed_extensions: Allowed file extensions. `['.*']` allows all.
     """
 
+    approved_hostnames: list[str] = field(
+        default_factory=lambda: ['localhost', '127.0.0.1']
+    )
```

Comment:
why not rely on the `uvicorn` `--host` flag?

Reply:
The uvicorn host flag is currently set to `0.0.0.0` by the default docker command (so that the server is accessible from outside the docker container). This means that unless docker is firewalled, anything within is it accessible on any network on which the machine is running. So, for example, if a person was running on a wifi hotspot at an internet cafe, somebody at a nearby table could access their server and use their LLM credits assuming there were not other security precautions in place and that they could figure out the IP (Maybe just search the subnet for an open port 3000).